predicators/ground_truth_models/kitchen/options.py

```diff
@@ -94,16 +94,47 @@ def policy(state: State, memory: Dict, objects: Sequence[Object],
     @classmethod
     def _create_terminal(cls, name: str,
                          operator: STRIPSOperator) -> ParameterizedTerminal:
-        del name
+        max_step_count = 25
 
         def terminal(state: State, memory: Dict, objects: Sequence[Object],
                      params: Array) -> bool:
             """Terminate when the option's corresponding operator's effects
             have been reached."""
-            del memory, params
-            grounded_op = operator.ground(tuple(objects))
-            if all(e.holds(state) for e in grounded_op.add_effects):
+            if "step_count" in memory:
+                step_count = memory["step_count"]
+            else:
+                step_count = 0
+
+            if "MoveTo" in operator.name:
+                if len(objects) == 2:
+                    gripper, _ = objects
+                else:
+                    assert len(objects) == 3
+                    gripper, _, _ = objects
+
+                gx = state.get(gripper, "x")
+                gy = state.get(gripper, "y")
+                gz = state.get(gripper, "z")
+
+                assert name.lower() == "move_delta_ee_pose"
+                if "target_pose" not in memory:
+                    memory["target_pose"] = np.array(
+                        [params[0], params[1], params[2]])
+
+                if np.allclose(np.array([gx, gy, gz]),
+                               memory["target_pose"],
+                               atol=0.2):
+                    return True
+            else:
+                grounded_op = operator.ground(tuple(objects))
```

Comment:
You or I should add comments here saying the during MoveTo we terminate based on getting to the initial sampled location and in every other option we terminate during add_effect completion.